SalesforceSDK/Core/Net/HttpCall.cs

```diff
@@ -314,20 +323,24 @@ public static HttpCall CreatePost(string url, string requestBody)
                 }
             }
             HttpResponseMessage message;
+
             try
             {
                 message = await _webClient.SendAsync(req);
-                HandleMessageResponse(message);
             }
-            catch (Exception ex)
+            catch (HttpRequestException ex)
             {
-                _httpCallErrorException = ex;
+                _httpCallErrorException = new DeviceOfflineException("Request failed to send, most likely because we were offline", ex);
```

Comment:
if the request is bad, the response will come back with an error which will through a HttpRequestException when EnsureStatusCode() is called on the response (in HandleResponseAsync) at that point we will catch the exception and return the error on the response.